### ISSUE TITLE
Player becoming unavailable and impacting other services

### ISSUE THREAD

**Issue by Thrasher2020** [nagyrobi/media_player.linkplay#80]
HASSOS 8.1
HA 2022.06.4

I am having an issue with the integration - I see a lot of messages in the logs.:

`2022-06-10 16:07:00 WARNING (MainThread) [homeassistant.components.media_player] Updating linkplay media_player took longer than the scheduled update interval 0:00:03`

`2022-06-10 16:16:27 WARNING (MainThread) [custom_components.linkplay.media_player] Failed communicating with LinkPlayDevice (httpapi) 'Office': <class 'asyncio.exceptions.TimeoutError'>`

Eventually it will cause other things to start timing out - I removed the integration for 24 hours and the issue disappeared. I'm really not sure what the problem could be.

![image](https://user-images.githubusercontent.com/65713394/173098069-17b569d5-1d13-4f60-9fd4-7d637981f7e7.png)

**Comment by nagyrobi** [comment]
Flaky network.

**Comment by Thrasher2020** [comment]
> Flaky network.

I would love to agree - however after much reconfiguring of my Unifi network (and then restoring it) I can't see how it can be the network. There are now two things affected when I reinstall the latest linkplay integration (3.1.5 doesn't cause this issue) - namely a PS5 python script that uses UDP to check the state of the PS5 and the WLED integration. 

Happy to be proven wrong of course as I have no idea what your integration does under the skin.

For now I'll leave it at 3.1.5 as that is not causing issues.

**Comment by nagyrobi** [comment]
I'm sorry but if it's a timeout error it has nothing to do with the component. The player doesn't respond...

**Comment by Thrasher2020** [comment]
Hi - as I said it works perfectly on 3.1.5 so I will just stay on this version. You can close this, maybe I'll try again when the next release is out.

**Comment by RosensRauk** [comment]
I'm having the same issues :/ help with trouble shooting?
I have run out of ideas. Two Audio Pro A10.

Logger: custom_components.linkplay.media_player
Source: custom_components/linkplay/media_player.py:434
Integration: Linkplay (documentation, issues)
First occurred: 19:43:13 (584 occurrences)
Last logged: 21:25:07

Failed async communicating with LinkPlayDevice (httpapi) 'orange': <class 'aiohttp.client_exceptions.ClientConnectorError'>
Failed async communicating with LinkPlayDevice (httpapi) 'lemon': <class 'aiohttp.client_exceptions.ClientConnectorError'>

Logger: homeassistant.components.media_player
Source: helpers/entity_platform.py:724
Integration: Mediaspelare (documentation, issues)
First occurred: 19:43:12 (297 occurrences)
Last logged: 21:25:06

Updating linkplay media_player took longer than the scheduled update interval 0:00:03
Updating dlna_dmr media_player took longer than the scheduled update interval 0:00:10

**Comment by Thrasher2020** [comment]
Linkplay is the common component for me. I did raise an issue but the developer said it couldn't be his integration.....

**Comment by molly200** [comment]
Hi all,

This is to confirm the above reports and that I'm having the same problems with the current version!

For me it started a few versions back, at about the time that I've tried the Music Assistant integration. 
I also run HA in docker and never had problems with linkplay, until the major HA breaking changes. 

BTW, the MA integration never worked properly for me and created a lot of instability and system breaks. Had to completely remove it and restore the system from backups, in order to have it work properly again. 

I noticed that the Linkplay integration is no longer listed under the Devices and Services integrations. It was helpful before, as I could monitor and access the devices and their status faster.  Unless I'm missing something,  the players were listed as Linkplay devices.

With the last three versions, the Linkplay integration does not play TTS or music from any source. As soon as TTS is selected, or a song is selected to be played, the players seem to disappear and are displayed as unavailable. After a while the players seem to intermittently reappear and disappear.   

As @[Thrasher2020 mentioned above, version 3.1.5 seems to be stable and working for me too.

Other things I've noticed for me, is that the HA DLNA media render integration seems to restore the sources links back to normal, after integrating the discovered players into HA. It's like the playlink integration depends on DLNA.  

Thanks for your time and work on this integration!
